.github/workflows/documentation.yml

```diff
@@ -45,20 +53,29 @@ jobs:
         key: ${{ runner.os }}-poetry-3.6-non-full-${{ hashFiles('**/poetry.lock') }}
         restore-keys: ${{ runner.os }}-poetry-3.6-non-full
 
+    - name: Load Yarn Cached Packages ⬇
+      uses: actions/cache@v1
+      with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-12.x-${{ hashFiles('newdocs/yarn.lock') }}
+        restore-keys: ${{ runner.os }}-yarn-12.x
+
     - name: Install Dependencies 📦
-      run: make install
-
-    # FIXME: https://github.com/RasaHQ/rasa/issues/6166
-    # - name: Build & Publish Docs 🏃‍♀️
-    #   env:
-    #     GITHUB_DOCS_KEY: ${{ secrets.GITHUB_DOCS_KEY }}
-    #   run: |
-    #     eval "$(ssh-agent -s)"; touch docs_key; chmod 0600 docs_key
-    #     echo "$GITHUB_DOCS_KEY" > "docs_key"
-    #     ssh-add docs_key
-
-    #     git config --global user.email "builds@github-ci.com"
-    #     git config --global user.name "GitHub CI"
-    #     git remote set-url --push origin "git@github.com:${{github.repository}}"
-
-    #     poetry run sphinx-versioning push docs docs . -- -b dirhtml -A html_theme=rasabaster
+      run: make install install-docs
+
+    - name: Build Docs variables 🧶
+      run: cd $DOCS_FOLDER && yarn run variables
+
+    - name: Build & Publish Docs 🏃‍♀️
+      env:
+        GITHUB_DOCS_KEY: ${{ secrets.GITHUB_DOCS_KEY }}
+      run: |
+        eval "$(ssh-agent -s)"; touch docs_key; chmod 0600 docs_key
+        echo "$GITHUB_DOCS_KEY" > "docs_key"
+        ssh-add docs_key
+
+        git config --global user.email "builds@github-ci.com"
+        git config --global user.name "GitHub CI"
+        git remote set-url --push origin "git@github.com:${{github.repository}}"
+
+        ./scripts/push_docs_to_branch.sh
```

Comment:
I still need to run this in the context of GH actions

Reply:
7a9fb8532d9

Reply:
I ran the workflow for these use cases (and iteratively fixing the bugs):
- push to master
- push to version but not major
- push to major version

For each, the Netlify deployment worked properly. The current latest commit in the [`documentation` branch](https://github.com/RasaHQ/rasa/tree/documentation) is the one with a new version, that way we see how it looks like in terms of folder structure / website experience. I'll cleanup the branch history after the review

Reply:
"next" version is available at this URL: https://rasahq-docsv2-rasa.netlify.app/next/
<img width="514" alt="Screenshot 2020-07-16 at 09 21 27" src="https://user-images.githubusercontent.com/1931615/87639615-ca5d3b80-c745-11ea-94e0-94ede9976ca2.png">

Default version is the latest version pushed (2.0.0 in this test case):
<img width="678" alt="Screenshot 2020-07-16 at 09 21 08" src="https://user-images.githubusercontent.com/1931615/87639619-ccbf9580-c745-11ea-8b94-dc220d4f56b1.png">

Reply:
regarding versions: 
- versions don't show up on the docs (i.e. there is no way to select a version). is there some configuration missing, or did the docosaurus docs site do something special?
<img width="395" alt="Bildschirmfoto 2020-07-16 um 10 09 45" src="https://user-images.githubusercontent.com/1098412/87644489-7f92f200-c74c-11ea-907d-821257485dac.png">
- would be good to have the version banner that docosaurus uses as well, but looks like they added that to their custom theme, so seems it needs a bit more effort and we should add it to the list of things to do later 
https://github.com/facebook/docusaurus/blob/5ccd24cc1f07efbd4e7c3fe68937e357c0e44855/packages/docusaurus-theme-classic/src/theme/DocItem/index.tsx#L126

<img width="904" alt="Bildschirmfoto 2020-07-16 um 10 19 14" src="https://user-images.githubusercontent.com/1098412/87645473-d220de00-c74d-11ea-9b0b-9ece3efe90da.png">

Reply:
Yes I think both are custom made. It's doable via [the theme](https://v2.docusaurus.io/docs/theme-classic#navbar-dropdown), and I guess we would only display the versions we want to display in this dropdown. Here is how Docusaurus [does it on their website](https://github.com/facebook/docusaurus/blob/v2.0.0-alpha.58/website/docusaurus.config.js#L113-L134).

Reply:
version dropdown added in 6fab2bfd17f and I updated the desc of https://github.com/RasaHQ/growth/issues/1501 to handle the version banner